docs/messaging/supplementary-guidelines/migration-guide.md

```diff
@@ -0,0 +1,434 @@
+# Messaging semantic convention stability migration guide
+
+Due to the significant number of modifications and the extensive user base
+affected by them, existing messaging instrumentations published by
+OpenTelemetry are required to implement a migration plan that will assist users in
+transitioning to the stable messaging semantic conventions.
+
+Specifically, when existing messaging instrumentations published by OpenTelemetry are
+updated to the stable messaging semantic conventions, they:
+
+- SHOULD introduce an environment variable `OTEL_SEMCONV_STABILITY_OPT_IN` in
+  their existing major version, which accepts:
+  - `messaging` - emit the stable messaging conventions, and stop emitting
+    the old messaging conventions that the instrumentation emitted previously.
+  - `messaging/dup` - emit both the old and the stable messaging conventions,
+    allowing for a phased rollout of the stable semantic conventions.
+  - The default behavior (in the absence of one of these values) is to continue
+    emitting whatever version of the old messaging conventions the
+    instrumentation was emitting previously.
+- Need to maintain (security patching at a minimum) their existing major version
+  for at least six months after it starts emitting both sets of conventions.
+- May drop the environment variable in their next major version and emit only
+  the stable messaging conventions.
+
+## Summary of changes
+
+This section summarizes the changes made to the messaging semantic conventions
+from
+[v1.24.0](https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/messaging/README.md).
+to
+[v1.TODO (stable)](https://github.com/open-telemetry/semantic-conventions/blob/v1.TODO/docs/messaging/README.md).
+
+### Common messaging attributes
+
+#### New attributes
+
+<!-- prettier-ignore-start -->
+| New attributes |
+| -------------- |
+| `messaging.destination.partition.id` |
+| `messaging.operation.name` |
+| `error.type` |
+| `server.port` |
+<!-- prettier-ignore-end -->
+
+#### Changed attributes
+
+<!-- prettier-ignore-start -->
+| Attribute change | Comments |
+| ---------------- | -------- |
+| `messaging.system` from string to enum |  |
+| `messaging.operation` → `messaging.operation.type` |  |
+| `messaging.client_id` → `messaging.client.id` |  |
+<!-- prettier-ignore-end -->
+
+### Removed attributes
+
+<!-- prettier-ignore-start -->
+| Removed attributes |
+| -------------- |
+| `network.transport` |
+| `network.type` |
+| `network.protocol.name` |
+| `network.protocol.version` |
+<!-- prettier-ignore-end -->
+
+### Kafka attributes
+
+#### New attributes
```

Comment:
those are not really new, but they were implicit before and now are explicit. 

Maybe we can add a common sentence that system-specific conventions now list all attributes explicitly and therefore instrumentation authors need to review corresponding documents to see if there are any new attributes that need to be populated?